plugins/src/main/java/opengrok/auth/plugin/ldap/LdapServer.java

```diff
@@ -122,13 +129,89 @@ public void setInterval(int interval) {
         this.interval = interval;
     }
 
+    private String urlToHostname(String urlStr) throws URISyntaxException {
+        URI uri = new URI(urlStr);
+        return uri.getHost();
+    }
+
+    /**
+     * This method converts the scheme from URI to port number.
+     * It is limited to the ldap/ldaps schemes.
+     * The method could be static however then it cannot be easily mocked in testing.
+     * @return port number
+     * @throws URISyntaxException if the URI is not valid
+     */
+    public int getPort() throws URISyntaxException {
```

Comment:
There is no reasonable way to get the functionality of `getservbyname()` libc library call: https://stackoverflow.com/questions/3645405/get-port-number-for-service-name